### ISSUE TITLE
Tickets Create() Request Is Broken Because ObjectSerializer.php Returns Empty Object, Stripping POST Values

### ISSUE THREAD

**Issue by TheFerrettDeveloper** [HubSpot/hubspot-api-php#137]
**TLDR: The ObjectSerializer is broken, returning an empty array of POST values, which causes validation errors when you try to create a ticket with an empty POST request.** 

So if you try to create a ticket through the suggested code on the HubSpot API documentation:  

`        $properties = [
            "hs_pipeline" => "1",
            "hs_pipeline_stage" => "1",
            "hs_ticket_priority" => "MEDIUM",
            "hubspot_owner_id" => "11",
            "subject" => "Test", 
        ];
`
`

        $SimplePublicObjectInput = new SimplePublicObjectInput(["properties" => $properties]);

        try {
            $apiResponse = $this->api->crm()->tickets()->basicApi()->create($SimplePublicObjectInput);
            var_dump($apiResponse);
        } catch (\Exception $e) {
            var_dump($e->getResponseObject());  
            die("Exception when calling basic_api->create: " . $e->getMessage());
        }`

Which is confusing, because it seems to have the necessary values.  

But if you dig deeper into the code, the stack trace goes through:

vendor/hubspot/api-client/codegen/Crm/Tickets/Api/BasicApi.php->create()
vendor/hubspot/api-client/codegen/Crm/Tickets/Api/BasicApi.php->createWithHttpInfo();
vendor/hubspot/api-client/codegen/Crm/Tickets/Api/BasicApi.php->createRequest() 
vendor/hubspot/api-client/codegen/Crm/Tickets/ObjectSerializer.php::sanitizeForSerialization()

And when sanitizeForSerialization() is passed a $simple_public_object_input, it attempts to read each of the fields like so: 

`                foreach($data as $property => $value) {
                    $values[$property] = self::sanitizeForSerialization($value);
                }`

Which is bad, because this particular $data actually has protected values, so it doesn't loop through each of the properties at all and returns an empty object/array.  

Any idea when this can be fixed?

**Comment by BatunaAz** [comment]
PASSED property=> $properties = [
    "hs_pipeline" => "0",
    "hs_pipeline_stage" => "1",
    "hs_ticket_priority" => "HIGH",
    "hubspot_owner_id" => "151284916",
    "subject" => "troubleshoot report"
];

 API RESPONSE=>

string(225) "{"status":"error","message":"Error creating TICKET.  Some required properties were not set.","correlationId":"c62da5a5-82a2-4128-b9d3-5503810393ea","context":{"properties":["hs_pipeline_stage"]},"category":"VALIDATION_ERROR"}"
  ["responseHeaders":protected]=>

  object(HubSpot\Client\Crm\Tickets\Model\Error)#31 (1) {
    ["container":protected]=>
    array(7) {
      ["message"]=>
      string(62) "Error creating TICKET.  Some required properties were not set."
      ["correlation_id"]=>
      string(36) "c62da5a5-82a2-4128-b9d3-5503810393ea"
      ["category"]=>
      string(16) "VALIDATION_ERROR"
      ["sub_category"]=>
      NULL
      ["errors"]=>
      NULL
      ["context"]=>
      array(1) {
        ["properties"]=>
        array(1) {
          [0]=>
          string(17) "hs_pipeline_stage"
        }
      }
      ["links"]=>
      NULL
    }
  }

Please can you reply my issue hurry. 

Thanks  a lot.

**Comment by ksvirkou-hubspot** [comment]
hi everyone 
Workable example for php SDK (8.2.0)
```php
$properties = [
    "hs_pipeline" => "0",
    "hs_pipeline_stage" => "1",
    "hs_ticket_priority" => "MEDIUM",
    "hubspot_owner_id" => "*",
    "subject" => "troubleshoot report"
];

$SimplePublicObjectInput = new SimplePublicObjectInput(["properties" => $properties]);

try {
    $apiResponse = $hubspot->crm()->tickets()->basicApi()->create($SimplePublicObjectInput);
    var_dump($apiResponse);
} catch (\Exception $e) {
    var_dump($e->getResponseObject());  
    die("Exception when calling basic_api->create: " . $e->getMessage());
}
```

**Comment by arivelli** [comment]
Hello Folks,

I have the same issue by trying to create a deal, I'm using the version 8.3.0 and I think the issue is on the line 85 of the file `vendor/hubspot/api-client/codegen/Crm/Deals/ObjectSerializer.php`

`$data` on that context is a `SimplePublicObjectInput` object which implements ModelInterface but somehow it isn't an instance of `ModelInterface` so the condition` if ($data instanceof ModelInterface) { `is always false

**Comment by ksvirkou-hubspot** [comment]
> Hello Folks,
> 
> I have the same issue by trying to create a deal, I'm using the version 8.3.0 and I think the issue is on the line 85 of the file `vendor/hubspot/api-client/codegen/Crm/Deals/ObjectSerializer.php`
> 
> `$data` on that context is a `SimplePublicObjectInput` object which implements `ModelInterface` but somehow it isn't an instance of `ModelInterface` so the condition`if ($data instanceof ModelInterface) {`is always false

Hi @arivelli 
Could you give me your code, please?

**Comment by arivelli** [comment]
Sure... have in mind it doesn't matter what array I sent to `SimplePublicObjectInput` class
`			$SimplePublicObjectInput = new SimplePublicObjectInput([
				'properties' => [
					'customer_email'    => 'xxxxxxx@xxxxxxx.xxx',
					'customer_name'     => 'Xxxxxxxx Xxxxxxx',
					'dealname'          => 'xxxxxxxx',
					"pipeline"          => "default",
				],
            ]);

			$deal = $this->hubspot
				->crm()
				->deals()
				->basicApi()
				->create($SimplePublicObjectInput);`

**Comment by arivelli** [comment]
> > Hello Folks,
> > I have the same issue by trying to create a deal, I'm using the version 8.3.0 and I think the issue is on the line 85 of the file `vendor/hubspot/api-client/codegen/Crm/Deals/ObjectSerializer.php`
> > `$data` on that context is a `SimplePublicObjectInput` object which implements `ModelInterface` but somehow it isn't an instance of `ModelInterface` so the condition`if ($data instanceof ModelInterface) {`is always false
> 
> Hi @arivelli Could you give me your code, please?

Sure... have in mind it doesn't matter what array I sent to `SimplePublicObjectInput` class

`
$SimplePublicObjectInput = new SimplePublicObjectInput([
	'properties' => [
		'customer_email'    => 'xxxxxxx@xxxxxxx.xxx',
		'customer_name'     => 'Xxxxxxxx Xxxxxxx',
		'dealname'          => 'xxxxxxxx',
		"pipeline"          => "default",
	],
]);

$deal = $this->hubspot
	->crm()
	->deals()
	->basicApi()
	->create($SimplePublicObjectInput);
`

**Comment by ksvirkou-hubspot** [comment]
> > > Hello Folks,
> > > I have the same issue by trying to create a deal, I'm using the version 8.3.0 and I think the issue is on the line 85 of the file `vendor/hubspot/api-client/codegen/Crm/Deals/ObjectSerializer.php`
> > > `$data` on that context is a `SimplePublicObjectInput` object which implements `ModelInterface` but somehow it isn't an instance of `ModelInterface` so the condition`if ($data instanceof ModelInterface) {`is always false
> > 
> > 
> > Hi @arivelli Could you give me your code, please?
> 
> Sure... have in mind it doesn't matter what array I sent to `SimplePublicObjectInput` class
> 
> ```
> $SimplePublicObjectInput = new SimplePublicObjectInput([
> 	'properties' => [
> 		'customer_email'    => 'xxxxxxx@xxxxxxx.xxx',
> 		'customer_name'     => 'Xxxxxxxx Xxxxxxx',
> 		'dealname'          => 'xxxxxxxx',
> 		"pipeline"          => "default",
> 	],
> ]);
> 
> $deal = $this->hubspot
> 	->crm()
> 	->deals()
> 	->basicApi()
> 	->create($SimplePublicObjectInput);
> ```

I have commented properties customer_email and customer_name and It works correctly 
With these props I gets error `Property values were not valid: ...`
Workable example
```php
$deal = new SimplePublicObjectInput([
    'properties' => [
#        'customer_email'    => 'oleh@com.com',
#		'customer_name'     => 'Oleh',
		'dealname'          => 'Caling',
        "pipeline" => "default",
    ],
]);

$response = $hubspot->crm()->deals()->basicApi()->create($deal);
```

**Comment by arivelli** [comment]
> 

nevermind, It was my fault, I copy&pasted the doc's example and didn't realize there are different `SimplePublicObjectInput` classes (I was using this one `HubSpot\Client\Crm\Contacts\Model\SimplePublicObjectInput`  when I should use use `HubSpot\Client\Crm\Deals\Model\SimplePublicObjectInput` )
